Add p5.js 2.x Project Generator extension to libraries

IrtizaNasar opened the pull request:
Adds the p5.js 2.x Project Generator VSCode extension to the libraries section.

This extension:
- Generates new p5.js 2.0 projects with clean, minimal setup
- Provides online/offline library options  
- Includes autocomplete, IntelliSense, and parameter hints for p5.js functions
- Perfect for artists, designers, educators, and creative coders

Changes made:
- Added new "integrations" category to config.ts
- Created p5js-project-generator.yaml with extension details
- Added featured image for the extension

Extension links:
- GitHub: https://github.com/IrtizaNasar/p5-2.vscode
- VSCode Marketplace: https://marketplace.visualstudio.com/items?itemName=Irti.p5js-project-generator

Rahulm2310 commented:
@lmccart please review this PR and let me know if the alt texts need improvement. 🙂

lmccart commented:
looks great, thanks!

IrtizaNasar committed:
Add p5.js 2.x Project Generator extension to libraries

Added VSCode extension for generating p5.js 2.0 projects with autocomplete and IntelliSense support. Includes new 'integrations' category.

IrtizaNasar committed:
Fix YAML format to match library standards

ksen0 commented:
Thanks @IrtizaNasar ! At the moment with `npm run dev` I see that the category is added, but the title is not consistent with the others:

<img width="832" height="1019" alt="image" src="https://github.com/user-attachments/assets/234fe0b9-9131-4e5f-9644-4d42ca6529d4" />

The yaml format is correct as it is, but please add the capitalized string in English here: https://github.com/processing/p5.js-website/blob/2.0/src/content/ui/en.yaml#L77 

(If you would like to contribute to translation and if you know Korean, Hindi, Spanish, or Chinese (simplified), then you could also put in a separate PR translating the library categories - but this should be a separate PR because it will need review from translation steward(s).)

ksen0 commented on src/content/libraries/config.ts:
```diff
@@ -20,6 +20,7 @@ export const categories = [
   "networking",
   "export",
   "utils",
+  "integrations",
```
To make this show up as capitalized please add the mapping to the [en strings](https://github.com/processing/p5.js-website/blob/2.0/src/content/ui/en.yaml). Thank you!

IrtizaNasar replied:
Hi @ksen0, Just wanted to check if this was all good now?

IrtizaNasar committed:
Add capitalized display name for integrations category

Addresses feedback to make category title consistent with others

IrtizaNasar commented:
Thank you @ksen0! This has been updated now. I'd also love to update the VSCode Setup Tutorial by integrating this plugin within it as a separate PR.

> Thanks @IrtizaNasar ! At the moment with `npm run dev` I see that the category is added, but the title is not consistent with the others:
> 
> <img alt="image" width="832" height="1019" src="https://private-user-images.githubusercontent.com/1304340/513802745-234fe0b9-9131-4e5f-9644-4d42ca6529d4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMwMzM3MTcsIm5iZiI6MTc2MzAzMzQxNywicGF0aCI6Ii8xMzA0MzQwLzUxMzgwMjc0NS0yMzRmZTBiOS05MTMxLTRlNWYtOTY0NC00ZDQyY2E2NTI5ZDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMTNUMTEzMDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmM3Mzc2YzIyNzljMjdmZTEwZWU4NDcxMTMwOTZhOGNlYWJjYWFjZTIyNzgzNWE3MWI1OTYzNjkxMWZmZTg3MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.RlNIOLN0eN4ndlhegIT323B2N3-ncRDqxY82k6Vulro">
> The yaml format is correct as it is, but please add the capitalized string in English here: https://github.com/processing/p5.js-website/blob/2.0/src/content/ui/en.yaml#L77
> 
> (If you would like to contribute to translation and if you know Korean, Hindi, Spanish, or Chinese (simplified), then you could also put in a separate PR translating the library categories - but this should be a separate PR because it will need review from translation steward(s).)

IrtizaNasar commented:
Hi @ksen0 , I just wanted to check if there are still any pending issues with this?

ksen0 committed:
Merge branch '2.0' into 2.0

ksen0 commented:
Apologies for the delay @IrtizaNasar I will merge it!
